index.d.ts

```diff
@@ -96,6 +119,56 @@ declare namespace customPlugin {
   type Handler<Config extends ServiceConfig = ServiceConfig, Request extends RequestGeneric = RequestGeneric> = BasicHandler<Config, Request, void>
   type AsyncHandler<Config extends ServiceConfig = ServiceConfig, Request extends RequestGeneric = RequestGeneric> = BasicHandler<Config, Request, Promise<any>>
 
+  //
+  // HTTP CLIENT
+  //
+  interface HttpClientBaseOptions {
+    headers?: http.IncomingHttpHeaders,
+    timeout?: number,
+    cert?: string,
+    key?: string,
+    ca?: string,
+  }
+  interface BaseHttpClientResponse {
+    headers: http.IncomingHttpHeaders
+    statusCode: number
+  }
+  interface StreamResponse extends BaseHttpClientResponse {
+    payload: NodeJS.ReadableStream
+  }
+  interface JSONResponse<Payload = any> extends BaseHttpClientResponse {
+    payload: Payload
+  }
+  interface BufferResponse extends BaseHttpClientResponse {
+    payload: Buffer
+  }
+  type Response<Payload = any> = StreamResponse | JSONResponse<Payload> | BufferResponse
+  interface HttpClientProxy {
+    protocol: 'http' | 'https'
+    host: string
+    port: number
+    auth: {
+      username: string
+      password: string
+    }
+  }
+  interface HttpClientOptions extends HttpClientBaseOptions {
+    returnAs?: 'STREAM' | 'JSON' | 'BUFFER';
+    validateStatus?: (statusCode: number) => boolean;
+    isMiaHeaderInjected?: boolean;
+    errorMessageKey?: string;
+    proxy?: HttpClientProxy;
+    query?: Record<string, string>;
+  }
+
+  interface HttpClient {
+    get: <ResponseType = Response>(path: string, options?: HttpClientOptions) => Promise<ResponseType>
+    post: <ResponseType = Response>(path: string, body: any | Buffer | ReadableStream, options?: HttpClientOptions) => Promise<ResponseType>
+    put: <ResponseType = Response>(path: string, body: any | Buffer | ReadableStream, options?: HttpClientOptions) => Promise<ResponseType>
```

Comment:
Can you also extract the body type definition?

Reply:
done